File: smart_farm/src/api/cameraDevice.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface CameraDevice {
  id?: number
  deviceId: number
  plotId: number
  plotName?: string
  plotCode?: string
  userId?: number
  username?: string
  nickname?: string
  name: string
  streamProtocol?: string
  streamUrl?: string
  snapshotUrl?: string
  resolution?: string
  onlineStatus?: number
  direction?: string
  createTime?: string
}

export interface CameraDevicePayload {
  deviceId: number
  plotId: number
  name: string
  streamProtocol?: string
  streamUrl?: string
  snapshotUrl?: string
  resolution?: string
  onlineStatus?: number
  direction?: string
}

export interface UpdateCameraDevicePayload extends Partial<CameraDevicePayload> {
  id: number
}

export interface CameraDeviceListParams {
  plotId?: number
  plotName?: string
  name?: string
  streamProtocol?: string
  onlineStatus?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询监控设备列表。 */
export const listCameraDevices = (params: CameraDeviceListParams) => {
  return get<ApiResult<PageInfo<CameraDevice>>>(
    '/camera-device/list',
    params as unknown as Record<string, unknown>,
  )
}

/** 查询监控设备详情。 */
export const getCameraDeviceById = (id: number) => {
  return get<ApiResult<CameraDevice>>(`/camera-device/${id}`)
}

/** 新增监控设备。 */
export const addCameraDevice = (data: CameraDevicePayload) => {
  return post<ApiResult<CameraDevice>>(
    '/camera-device/add',
    data as unknown as Record<string, unknown>,
  )
}

/** 编辑监控设备。 */
export const updateCameraDevice = (data: UpdateCameraDevicePayload) => {
  return http<ApiResult<CameraDevice>>({ url: '/camera-device', method: 'put', data })
}

/** 删除单个监控设备。 */
export const deleteCameraDevice = (id: number) => {
  return http<ApiResult<void>>({ url: `/camera-device/${id}`, method: 'delete' })
}

/** 批量删除监控设备。 */
export const batchDeleteCameraDevices = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/camera-device/batch', method: 'delete', data: ids })
}

/** 修改在线状态，1 表示在线，0 表示离线。 */
export const updateCameraDeviceOnlineStatus = (id: number, onlineStatus: number) => {
  return http<ApiResult<void>>({
    url: `/camera-device/${id}/online-status`,
    method: 'put',
    params: { onlineStatus },
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/CameraDevice.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 监控设备实体类。
 *
 * <p>该类与 camera_device 表对应，用于承载地块摄像头和视频流基础配置。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CameraDevice {

    /** 监控设备主键ID。 */
    private Long id;

    /** 原 iot_device.id 或外部设备编号，用于保留摄像头来源设备标识。 */
    private Long deviceId;

    /** 所属地块ID。 */
    private Long plotId;

    /** 所属地块名称，列表和详情查询时通过 plot 表联查得到。 */
    private String plotName;

    /** 所属地块编号，列表和详情查询时通过 plot 表联查得到。 */
    private String plotCode;

    /** 地块所属用户ID。 */
    private Long userId;

    /** 地块所属用户名。 */
    private String username;

    /** 地块所属用户昵称。 */
    private String nickname;

    /** 监控设备名称。 */
    private String name;

    /** 视频流协议，如 RTSP、GB28181、HTTP。 */
    private String streamProtocol;

    /** 视频流地址。 */
    private String streamUrl;

    /** 截图地址。 */
    private String snapshotUrl;

    /** 分辨率，如 1920x1080。 */
    private String resolution;

    /** 在线状态，0离线，1在线。 */
    private Integer onlineStatus;

    /** 监控方向。 */
    private String direction;

    /** 创建时间。 */
    private LocalDateTime createTime;
}
